moma_mission/demo/piloting.py

```diff
@@ -22,65 +24,139 @@
+                                       'Failure': 'Failure'})
+
+        rospy.loginfo("Grasp valve")
+        state_machine.add('GRASP_VALVE',
+                          PathVisitorState,
+                          transitions={'Completed': 'CLOSE_GRIPPER',
+                                       'Failure': 'DETECTION_DECISION'})
+
+        rospy.loginfo("Close gripper")
+        state_machine.add('CLOSE_GRIPPER',
+                          GripperControl,
+                          transitions={'Completed': 'MANIPULATE_VALVE',
+                                       'Failure': 'MANIPULATE_VALVE'})
+
+        rospy.loginfo("Manipulate valve")
+        state_machine.add('MANIPULATE_VALVE',
+                          PathVisitorState,
+                          transitions={'Completed': 'OPEN_GRIPPER',
+                                       'Failure': 'OPEN_GRIPPER'})
+
+        rospy.loginfo("Open gripper")
+        state_machine.add('OPEN_GRIPPER',
+                          GripperControl,
+                          transitions={'Completed': 'BACKOFF_VALVE',
+                                       'Failure': 'Failure'})
+
+        rospy.loginfo("Backoff valve")
+        state_machine.add('BACKOFF_VALVE',
+                          PathVisitorState,
+                          transitions={'Completed': 'HOMING_FINAL',
+                                       'Failure': 'Failure'})
 
         homing_sequence_final = StateMachineRos(
             outcomes=['Success', 'Failure'])
 
         with homing_sequence_final:
-            homing_sequence_final.add('HOME_ROBOT_FINAL', JointsConfigurationAction, transitions={'Completed': 'Success',
-                                                                                                  'Failure': 'Failure'})
-
-        state_machine.add('HOMING_FINAL', homing_sequence_final, transitions={'Success': 'Success',
-                                                                              'Failure': 'Failure'})
+            rospy.loginfo("Home robot final")
+            homing_sequence_final.add('HOME_ROBOT_FINAL',
+                                      JointsConfigurationAction,
+                                      transitions={'Completed': 'Success',
+                                                   'Failure': 'Failure'})
+
+        state_machine.add('HOMING_FINAL',
+                          homing_sequence_final,
+                          transitions={'Success': 'Success',
+                                       'Failure': 'Failure'})
 except Exception as exc:
     rospy.logerr(exc)
     sys.exit(0)
 
 # Create and start the introspection server
-introspection_server = smach_ros.IntrospectionServer(
-    'piloting_mission_server', state_machine, '/mission_planner')
-introspection_server.start()
+# introspection_server = smach_ros.IntrospectionServer(
+#     'piloting_mission_server', state_machine, '/mission_planner')
+# introspection_server.start()
```

Comment:
The introspection server is what allows to visualize the state machine. Unfortunately this is somethign we cannot currently used because the visualizer is not supported in `noetic`. I originally removed it with the intention to check if it was impacting the startup delay (rather due to the ROS streaming). We can remove it as this is anyway dead code.